pkg/lightning/errormanager/errormanager.go

```diff
@@ -83,37 +85,51 @@ const (
-			row_data    text NOT NULL COMMENT 'decoded from raw_row, human readable only, not for machine use',
-			raw_key     mediumblob NOT NULL COMMENT 'the conflicted key',
-			raw_value   mediumblob NOT NULL COMMENT 'the value of the conflicted key',
-			raw_handle  mediumblob NOT NULL COMMENT 'the data handle derived from the conflicted key or value',
-			raw_row     mediumblob NOT NULL COMMENT 'the data retrieved from the handle',
-			is_data_kv  tinyint(1) NOT NULL,
+			is_precheck_conflict tinyint(1) NOT NULL DEFAULT 0,
+			task_id              bigint NOT NULL,
+			create_time          datetime(6) NOT NULL DEFAULT now(6),
+			table_name           varchar(261) NOT NULL,
+			index_name           varchar(128) NOT NULL,
+			key_data             text NOT NULL COMMENT 'decoded from raw_key, human readable only, not for machine use',
+			row_data             text NOT NULL COMMENT 'decoded from raw_row, human readable only, not for machine use',
+			raw_key              mediumblob NOT NULL COMMENT 'the conflicted key',
+			raw_value            mediumblob NOT NULL COMMENT 'the value of the conflicted key',
+			raw_handle           mediumblob NOT NULL COMMENT 'the data handle derived from the conflicted key or value',
+			raw_row              mediumblob NOT NULL COMMENT 'the data retrieved from the handle',
+			is_data_kv           tinyint(1) NOT NULL,
 			INDEX (task_id, table_name),
 			INDEX (index_name),
-			INDEX (table_name, index_name)
+			INDEX (table_name, index_name),
+			INDEX (is_precheck_conflict)
 		);
 	`
 
-	createDupRecordTable = `
-		CREATE TABLE IF NOT EXISTS %s.` + DupRecordTable + ` (
-			task_id     bigint NOT NULL,
-			create_time datetime(6) NOT NULL DEFAULT now(6),
-			table_name  varchar(261) NOT NULL,
-			path        varchar(2048) NOT NULL,
-			offset      bigint NOT NULL,
-			error       text NOT NULL,
-			row_id 	    bigint NOT NULL COMMENT 'the row id of the conflicted row',
-			row_data    text NOT NULL COMMENT 'the row data of the conflicted row',
-			KEY (task_id, table_name)
+	createDupRecordTableName = `
+		CREATE TABLE IF NOT EXISTS %s.` + DupRecordTableName + ` (
+			is_precheck_conflict tinyint(1) NOT NULL DEFAULT 1,
+			task_id              bigint NOT NULL,
+			create_time          datetime(6) NOT NULL DEFAULT now(6),
+			table_name           varchar(261) NOT NULL,
+			path                 varchar(2048) NOT NULL,
+			offset               bigint NOT NULL,
+			error                text NOT NULL,
+			row_id 	             bigint NOT NULL COMMENT 'the row id of the conflicted row',
+			row_data             text NOT NULL COMMENT 'the row data of the conflicted row',
+			INDEX (task_id, table_name),
+			INDEX (is_precheck_conflict)
 		);
 	`
 
+	createConflictView = `
+    	CREATE OR REPLACE VIEW  %s.` + ConflictViewName + `
+			AS SELECT is_precheck_conflict, task_id, create_time, table_name, index_name, key_data, row_data,
```

Comment:
can we use `select 0 as is_precheck_conflict, ..... from xxx`? so we don't have maintain it in each table

Reply:
modified